cmd/swapd/main.go

```diff
@@ -262,28 +262,21 @@ func setLogLevels(level string) {
 }
 
 func runDaemon(c *cli.Context) error {
+	ctx, cancel := context.WithCancel(c.Context)
+	defer cancel()
+	go signalHandler(ctx, cancel)
```

Comment:
The function `d.wait()` wasn't waiting and was only interacting with the context, so I decoupled it from the daemon, renamed it to `signalHandler`, and invoke it earlier.